deps/chakrashim/lib/chakra_debug.js

```diff
@@ -0,0 +1,1289 @@
+// of this software and associated documentation files(the 'Software'), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and / or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+
+/* eslint-disable strict */
+(function(globalObject, keepAliveObject, traceDebugJson) {
+  var globalExecutionState;
+  var pendingMessages = [];
+
+  var Logger = (function() {
+    var logTypes = {
+      API: 1,
+      DebugJson: 2,
+      Error: 3
+    };
+
+    var enabledLogType = [];
+    enabledLogType[logTypes.API] = false; // For development purpose
+    enabledLogType[logTypes.DebugJson] = traceDebugJson === true;
+    enabledLogType[logTypes.Error] = true; // Always enabled
+
+    function logFunc(logType, argsArray) {
+      if (enabledLogType[logType]) {
+        var printStr = '';
+        for (var index in argsArray) {
+          var obj = argsArray[index];
+          if (obj != undefined) {
+            printStr += (typeof obj != 'string') ? JSON.stringify(obj) : obj;
+            printStr += ': ';
+          }
+        }
+        chakraDebug.log(printStr);
+      }
+    }
+
+    return {
+      LogAPI: function(msg, obj) {
+        logFunc(logTypes.API, arguments);
+      },
+      LogDebugJson: function(msg, obj) {
+        logFunc(logTypes.DebugJson, arguments);
+      },
+      LogError: function(msg, obj) {
+        logFunc(logTypes.Error, arguments);
+      }
+    };
+  })();
+
+  function CallHostFunctionNoLog(fn) {
+    var args = [].slice.call(arguments, 1);
+    var start = new Date();
+    var result = fn.apply(undefined, args);
+    var end = new Date();
+    Logger.LogAPI(`${fn.name}(${args}):${end - start}`);
+    return result;
+  }
+
+  function CallHostFunction(fn) {
+    var args = [].slice.call(arguments, 1);
+    var start = new Date();
+    var result = fn.apply(undefined, args);
+    var end = new Date();
+    Logger.LogAPI(`${fn.name}(${args}):${end - start}`, result);
+    return result;
+  }
+
+  if (!globalObject.Debug) {
+    Object.defineProperty(globalObject, 'Debug', {
+      value: {}, enumerable: false, configurable: false, writable: false
```

Comment:
The property descriptor keys are set to their default values- why not just do globalObject.Debug = {}